src/dbus_api/util.rs

```diff
@@ -194,3 +204,23 @@ pub fn thread_safe_to_dbus_sendable(ia: InterfacesAddedThreadSafe) -> Interfaces
         })
         .collect()
 }
+
+pub fn poll_exit_and_future<E, F, R>(exit: E, future: F) -> StratisResult<Option<R>>
+where
+    E: Future<Output = Result<(), RecvError>>,
+    F: Future<Output = R>,
+{
+    pin_mut!(exit);
+    pin_mut!(future);
+
+    match block_on(select(exit, future)) {
+        Either::Left((Ok(()), _)) => {
+            info!("D-Bus tree handler was notified to exit");
+            Ok(None)
+        }
+        Either::Left((Err(_), _)) => Err(StratisError::Msg(
+            "D-Bus tree handler can no longer be notified to exit; shutting down...".to_string(),
```

Comment:
Yes, you're right. Would you prefer to pass in an error message as an argument? That way we can provide specificity while also using this elsewhere if needed.

Reply:
Yes, that is probably better.

Reply:
I think that making the message more general is probably better here. After reviewing the error message, it's returned by the exit future failure and is able to be generalized fairly easily to avoid passing in an error message.